prime-dotnet-webapi/Services/AgreementService.cs

```diff
@@ -117,7 +117,7 @@ public async Task CreateEnrolleeAgreementAsync(int enrolleeId)
             var agreement = new Agreement
             {
                 EnrolleeId = enrolleeId,
-                AgreementVersionId = await FetchNewestAgreementVersionIdOfType(dto.NewestAssignedAgreement.Value),
+                AgreementVersionId = (await FetchNewestAgreementVersionOfTypeAsync(dto.NewestAssignedAgreement.Value)).Id,
```

Comment:
Do you know that this will always have an `Id`?  You'll either need to maintain both methods or perform a null check.